### DISCUSSION TITLE
Handling Ordering app use case

### DISCUSSION THREAD

**Discussion by merabtenei** [pocketbase/pocketbase#827] [Q&A]
Hi everyone,
I was about to start a project with pocketbase but there's some very basic use case I can't see pocketbase covering. It's a simple making an Order with multiple Items use case.
We can have an Order collection :
```
orderID: string
orderDate: DateTime
```
And an Items collection with Item having a reference to Order unique ID:
```
OrderItemID: string
OrderID: string
ItemName: string
ItemQty: integer
ItemPrice: float
```
The problem is when creating the Order, there isn't any way to create multiple documents at once. For an Order with 10 Items I need to create the Order document and then call **client.records.create 10 times for each OrderItem** ... What happens if there's an error for some reason when creating the 5th document (network failure for example) We'll have an Order with 5 items when it is supposed to have 10, and there's no way to revert the insertion and throw an error like in DB transactions. This is really a bottleneck that is preventing me from using pocketbase for a serious app.

Any suggestions ? How would you deal with this use case using Pocketbase.

**Comment by ganigeorgiev** [comment]
Transactional bulk create/update/delete is in the [roadmap](https://github.com/orgs/pocketbase/projects/2) and will be handled after v0.8 as part of #48.

> How would you deal with this use case using Pocketbase.

It depends on the desired workflow. Personally, I would just create the items one by one and add a status field that will indicate complete/incomplete order. This way the user will also have an option to continue from the last state before the network error.

If you really need to execute a bulk transactional operation, then you can extend PocketBase and define your own custom endpoint using the [event hooks](https://pocketbase.io/docs/event-hooks/). To send a request to your new endpoint you can use the SDK helper `client.send(path, { /* requestOptions */ })`.

**Reply by merabtenei** [reply]
Thanks for the quick answer. I though of that solution, adding a status field on the Order collection isn't a bad Idea.
